src/ucp/rndv/rndv.c

```diff
@@ -1343,41 +1329,25 @@ static void ucp_rndv_send_frag_rtr(ucp_worker_h worker, ucp_request_t *rndv_req,
         freq->recv.buffer                 = mdesc->ptr;
         freq->recv.datatype               = ucp_dt_make_contig(1);
         freq->recv.mem_type               = mdesc->memh->mem_type;
+        freq->recv.state.dt.contig.memh   = mdesc->memh;
         freq->recv.length                 = frag_size;
-        freq->recv.state.dt.contig.md_map = 0;
         freq->recv.frag.offset            = offset;
         freq->flags                       = UCP_REQUEST_FLAG_RNDV_FRAG;
 
         ucp_request_set_super(freq, rreq);
 
-        alloc_md_map = 0;
-        /* TODO: Check if can avoid for inter-node eps */
-        ucs_for_each_bit(md_index, mdesc->memh->md_map) {
-            if (md_index == mdesc->memh->alloc_md_index) {
-                md_attr = &context->tl_mds[md_index].attr;
-                if (md_attr->cap.flags & UCT_MD_FLAG_RKEY_PTR) {
-                    alloc_md_map |= UCS_BIT(md_index);
-                }
-                break;
```

Comment:
seems this logic was not preserved because it checks for UCT_MD_FLAG_RKEY_PTR and now code isn't

Reply:
it's something GPU specific. let's try to preserve the logic.
@bureddy and @Akshay-Venkatesh  may be able to provide more details